File: RedSheet.DbEntities/Models/vOurTheirRequirenmentDetail.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vOurTheirRequirenmentDetails", Schema = "dbo")]
	[Serializable]
	public partial class vOurTheirRequirenmentDetail  
	{
 
	[Required]
	public bool IsZoma { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string OurLDO { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string OurMDO { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int OurRequirementDetailId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string OurRequirenment { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectRequirementId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int RequirementCategoryId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string TheirLDO { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string TheirMDO { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int TheirRequirementDetailId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string TheirRequirenment { get; set; } 

	public vOurTheirRequirenmentDetail()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-our-their-requirenment-detail.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vOurTheirRequirenmentDetail {
    constructor(vOurTheirRequirenmentDetail?: vOurTheirRequirenmentDetail )  {
        let properties = [ "isZoma", "ourLDO", "ourMDO", "ourRequirementDetailId", "ourRequirenment", "projectRequirementId", "requirementCategoryId", "theirLDO", "theirMDO", "theirRequirementDetailId", "theirRequirenment",];
        for (let property of properties)
            if (vOurTheirRequirenmentDetail && vOurTheirRequirenmentDetail[property])
                this[property] = vOurTheirRequirenmentDetail[property];
    }
 
	isZoma : boolean = false ;
 
    @required()
    @maxLength(1000)
	ourLDO : string =   undefined;
 
    @required()
    @maxLength(1000)
	ourMDO : string =   undefined;
 
	ourRequirementDetailId : number =   0 ;
 
    @required()
    @maxLength(1000)
	ourRequirenment : string =   undefined;
 
    @range(1,2147483647)
	projectRequirementId : number =   undefined;
 
    @range(1,2147483647)
	requirementCategoryId : number =   undefined;
 
    @required()
    @maxLength(1000)
	theirLDO : string =   undefined;
 
    @required()
    @maxLength(1000)
	theirMDO : string =   undefined;
 
    @range(1,2147483647)
	theirRequirementDetailId : number =   undefined;
 
    @required()
    @maxLength(1000)
	theirRequirenment : string =   undefined;


}
